### ISSUE TITLE
Feature request: Search all podcast episodes by term

### ISSUE THREAD

**Issue by TSampley** [Podcastindex-org/docs-api#132]
I have looked over the API docs and found the ability to search for podcasts by term; however, AntennaPod has recently received a request to search for *episodes* by term across all podcasts – this is apparently a feature some other apps provide (presumably paid apps?)

In order to prevent crawling the entire API or spinning up an intermediate API to handle the indexing/caching required to satisfy this request, I would like to request the addition of an endpoint to search podcast-episodes by term.

If this feature already exists and I've simply overlooked the endpoint, apologies for wasting your time 🙏🏻

https://github.com/AntennaPod/AntennaPod/issues/7486

**Comment by daveajones** [comment]
Episode search by term comes up as a request a lot, but it's got a lot of problems.  You'd be searching description and title text, which returns so much noise.  There are about 160 million episodes in the Index so even searching for simple phrases like "european union" returns millions of episodes with no way to rank their relevance since we don't have first party data to give any weightings to anything.

I'd love to provide such a thing if the community could help me design how it would be more than just noise.  I'm open to ideas.

**Comment by TSampley** [comment]
I would naively suggest sorting by recency and only returning something like 30 results with a paging parameter. I personally don't understand that value of such a search function - I'm trying to get more user info on that.

Looping in original issue author for insight - @muctebanesiri, does sorting by latest episode make sense to you?

**Comment by muctebanesiri** [comment]
> I would naively suggest sorting by recency and only returning something like 30 results with a paging parameter. I personally don't understand the value of such a search function - I'm trying to get more user info on that.
> 
> Looping in original issue author for insight - @muctebanesiri, does sorting by latest episode make sense to you?

Let me share some screenshots of what I saw so maybe it would be more helpful.

![Imagepipe_0](https://github.com/user-attachments/assets/c53ae0bc-a2b7-48a3-b300-757669125c17)
This is a paid app as you correctly guessed with the feature in action. Episodes are not sorted by recency but I think sorting by recency would be more helpful.

![Imagepipe_1](https://github.com/user-attachments/assets/aa8a6143-06dd-44ec-84d9-2758b8a7a2e8)
The second app with the exact feature (even the sorting is identical) is actually a fork of AntennaPod.

**Comment by daveajones** [comment]
In order to Index cost effectively (I would need to spin up a new search server) we would probably need to Index only a fraction of the full episodes table.  If recency is a decent selector then maybe a rolling X number of months window so that we don't end up pulling in very old episode data?

I think we would also need a really good stop word list.

**Comment by keunes** [comment]
@daveajones A stop word list being a list with words to exclude from indexing like 'and'?
Here's a good starting point: https://www.ranks.nl/stopwords I'll send them an email to ask if we can use it.

If we're implementing a timeframe anyway (and have a limited, recent dataset), might we sort based on relevance? For work I'm on a project that allows you to search organisations, but you cannot search on relevance (only creation date, most recent) and it's… not great.

**Comment by muctebanesiri** [comment]
> @muctebanesiri Is that Podcini then? We might have a look where they're searching against (which data source).

No, it's FocusPodcast a more recent fork. https://github.com/allentown521/FocusPodcast

**Comment by keunes** [comment]
> IIRC, AntennaPod also uses iTunes as a data source but only for the top podcasts list

That's correct. Didn't know iTunes also enabled searching through episodes.
